projects/core/src/event/event.service.ts

```diff
@@ -64,40 +59,26 @@ export class EventService {
         );
       }
     } else {
-      event.sources$.next([...sources, source$]);
+      eventMeta.mergingSubject.add(source$);
     }
 
-    return () => this.unregister(eventType, source$);
-  }
-
-  /**
-   * Unregisters an event source for the given event type
-   *
-   * @param eventType the event type
-   * @param source$ an observable that represents the source
-   */
-  private unregister<T>(eventType: Type<T>, source$: Observable<T>): void {
-    const event = this.getEventMeta(eventType);
-    const newSources: Observable<T>[] = event.sources$.value.filter(
-      (s$) => s$ !== source$
-    );
-    event.sources$.next(newSources);
+    return () => eventMeta.mergingSubject.remove(source$);
   }
 
   /**
    * Returns a stream of events for the given event type
    * @param eventTypes event type
    */
   get<T>(eventType: Type<T>): Observable<T> {
-    return this.getEventMeta(eventType).output$;
+    let output$ = this.getEventMeta(eventType).mergingSubject.output$;
+    if (isDevMode()) {
+      output$ = this.getValidatedEventStream(output$, eventType);
```

Comment:
You have `share()` in output$, but you are validating here, thus one validation per subscriber, which might look funny in a console